funcx_endpoint/funcx_endpoint/endpoint/rabbit_mq/result_queue_publisher.py

```diff
@@ -0,0 +1,82 @@
+import logging
+
+import pika
+
+logger = logging.getLogger(__name__)
+
+
+class ResultQueuePublisher:
+    """ResultPublisher publishes results to a topic exchange, with the {endpoint_id}.results
+    as a routing key.
+    """
+
+    def __init__(
+        self,
+        endpoint_id: str,
+        pika_conn_params: pika.connection.Parameters,
+        exchange="results",
+    ):
+        """
+        Parameters
+        ----------
+        endpoint_uuid: str
+            Endpoint UUID string used to identify the endpoint
+        pika_conn_params: pika.connection.Parameters
+            Pika connection parameters to connect to RabbitMQ
+        exchange: str
+            Exchange name. Default: "result"
+        """
+        self.endpoint_id = endpoint_id
+        self.routing_key = f"{self.endpoint_id}.results"
+        self.params = pika_conn_params
+        self.params.heartbeat = 30
+        self.params.blocked_connection_timeout = 60
+        self.exchange = exchange
+        self.exchange_type = "topic"
+        self._channel = None
+        self._connection = None
+
+    def connect(self) -> pika.BlockingConnection:
+        """Connect
+
+        :rtype: pika.BlockingConnection
+        """
+        logger.info(f"Connecting to {self.params}")
+        self._connection = pika.BlockingConnection(self.params)
+        self._channel = self._connection.channel()
+        self._channel.confirm_delivery()
+        self._channel.exchange_declare(
+            exchange=self.exchange, exchange_type=self.exchange_type
+        )
+        # TO-DO: This shouldn't be done on the endpoint side
+        self._channel.queue_declare(queue="results")
```

Comment:
I've switched this over to passive for now. I suspect that when authz is setup on rabbitMQ, this might go wonky.

Reply:
> I suspect that when authz is setup on rabbitMQ, this might go wonky.

I'm curious why you think authz will have an impact here? I wouldn't expect any change to be needed to this call.

Reply:
Basically only the services would have perms for the `results` queue, I'm not sure if the result_publisher can declare it at all. It's probably fine, but something to keep in mind while setting access perms.

Reply:
`passive` is documented as meaning that the call only checks that the queue exists. I would be surprised if it breaks, but not shocked.

In any case, the change I wanted, switching to `passive` so that this is no longer doing queue creation, is in, so I'm marking this resolved.